docs/how-to-guides.md

```diff
@@ -430,3 +430,114 @@ If everything went well, you can remove the `bsconfig.json` file, and remove any
 dependencies needed by Melange from the `package.json`, as they will be
 appearing in the `opam` file instead, as it was mentioned in the
 [`bs-dependencies` section](#bs-dependencies).
+
+## Migrate
+
+This section contains information about migrating from older versions of Melange
+to newer ones.
+
+### To v2 from v1
+
+Melange v2 is only compatible with OCaml 5.1. In order to upgrade, let's update
+the local opam switch first, to make sure the local repository gets the versions
+v2 of Melange and 5.1 of OCaml:
+
+```bash
+opam update
+```
+
+Now, update the version of the OCaml compiler in the local switch to 5.1:
+
+```bash
+opam install --update-invariant ocaml-base-compiler.5.1.0
+```
+
+Finally, we can upgrade all packages to get Melange v2 and the latest version of
+all libraries:
+
+```bash
+opam upgrade
+```
+
+To make sure you have the latest version of Melange, you can use the `opam list`
+subcommand:
+
+```bash
+opam list --installed melange
+# Packages matching: name-match(melange) & installed
+# Name  # Installed    # Synopsis
+melange 2.0.0          Toolchain to produce JS from Reason/OCaml
+```
+
+Before building, we have to update some parts of the configuration to make it
+work with v2.
+
+#### `melange.ppx` includes now most syntax transformations
+
+Most of the attributes used to write bindings are now handled by `melange.ppx`.
+If you get errors of the kind `Unused attribute`, then you probably need to add
```

Comment:
this will also be true for "most weird errors you can't explain". it's probably just missing the ppx